driver-kotlin-extensions/src/main/kotlin/com/mongodb/kotlin/client/model/Projections.kt

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ * Copyright (C) 2016/2022 Litote
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @custom-license-header
+ */
+package com.mongodb.kotlin.client.model
+
+import com.mongodb.annotations.Beta
+import com.mongodb.annotations.Reason
+import com.mongodb.client.model.Aggregates
+import com.mongodb.client.model.Projections
+import kotlin.reflect.KProperty
+import kotlin.reflect.KProperty1
+import org.bson.conversions.Bson
+
+/** The projection of the property. This is used in an aggregation pipeline to reference a property from a path. */
+public val <T> KProperty<T>.projection: String
+    get() = path().projection
+
+/** The projection of the property. */
+public val String.projection: String
+    get() = "\$$this"
+
+/** In order to write `$p.p2` */
+@JvmSynthetic public infix fun <T0, T1> KProperty1<T0, T1?>.projectionWith(p2: String): String = "$projection.$p2"
+
+/**
+ * Creates a projection of a property whose value is computed from the given expression. Projection with an expression
+ * can be used in the following contexts:
+ * <ul>
+ * <li>$project aggregation pipeline stage.</li>
+ * <li>Starting from MongoDB 4.4, it's also accepted in various find-related methods within the {@code
+ *   MongoCollection}-based API where projection is supported, for example: <ul>
+ * <li>{@code find()}</li>
+ * <li>{@code findOneAndReplace()}</li>
+ * <li>{@code findOneAndUpdate()}</li>
+ * <li>{@code findOneAndDelete()}</li>
+ * </ul>
+ *
+ * </li> </ul>
+ *
+ * @param expression the expression
+ * @param <T> the expression type
+ * @return the projection
+ * @see #computedSearchMeta(String)
+ * @see Aggregates#project(Bson)
+ */
+@JvmSynthetic
+public infix fun <T> KProperty<T>.computedFrom(expression: Any): Bson =
```

Comment:
In Filters we added both extension methods (where possible) as well as simple methods eg: `computedFrom(property: KProperty<T>, expression: Any)`.  Although, this adds two ways of doing things, it does mean that users can read the driver docs and easily port the Java examples.

We should also add them here, to keep this inline with the `Filters` class.